src/sas/sascalc/file_converter/bsl_loader.py

```diff
@@ -99,33 +172,39 @@ def load_frames(self, frames):
 
         return frame_data
 
+    def load_data(self):
+        """
+        Loads the file named in filename in 4 byte float, in either
+        little or big Endian depending on self.swap_bytes.
+
+        :return: np array of loaded floats.
+        """
+        # Set dtype to 4 byte float, big or little endian depending on swap_bytes.
+        dtype = ('>f4', '<f4')[self.swap_bytes]
+
+        # Size of float as stored in binary file should be 4 bytes.
+        float_size = 4
 
-    def __setattr__(self, name, value):
-        if name == 'filename':
-            return self.set_filename(value)
-        elif name == 'n_frames':
-            return self.set_n_frames(value)
-        elif name == 'frame':
-            return self.set_frame(value)
-        elif name == 'n_pixels':
-            return self.set_n_pixels(value)
-        elif name == 'n_rasters':
-            return self.set_n_rasters(value)
-        elif name == 'swap_bytes':
-            return self.set_swap_bytes(value)
-        return CLoader.__setattr__(self, name, value)
-
-    def __getattr__(self, name):
-        if name == 'filename':
-            return self.get_filename()
-        elif name == 'n_frames':
-            return self.get_n_frames()
-        elif name == 'frame':
-            return self.get_frame()
-        elif name == 'n_pixels':
-            return self.get_n_pixels()
-        elif name == 'n_rasters':
-            return self.get_n_rasters()
-        elif name == 'swap_bytes':
-            return self.get_swap_bytes()
-        return CLoader.__getattr__(self, name)
+        offset = self.n_pixels * self.n_rasters * self.frame * float_size
+        load = np.empty(0, dtype=np.float64)
```

Comment:
Statement does nothing; `load` is replaced later in the code.  If there is an exception then `load` will not be used, so no need to create an empty one.

Also, `data` would be a better name for the loaded data.

Reply:
Changed.